cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizationService.java

```diff
@@ -0,0 +1,280 @@
+import java.util.Map;
+import java.util.Properties;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import java.util.zip.ZipException;
+
+/**
+ * Class to instantiate {@link Authorizer} extensions. Authorization extensions are instantiated using a
+ * separate {@link ClassLoader} that is built using a bundled jar for the {@link Authorizer} extension that
+ * contains all its required dependencies. The {@link ClassLoader} is created with the parent as the classloader with
+ * which the {@link Authorizer} interface is instantiated. This parent only has classes required by the
+ * {@code cdap-security-spi} module.
+ *
+ * It is implemented as a {@link AbstractIdleService} so it can manage initialization and destruction of the
+ * {@link AuthorizerClassLoader} in a reliable manner.
+ *
+ * The {@link AuthorizationService} has the following expectations from the extension:
+ * <ul>
+ *   <li>Authorization is enabled setting the parameter {@link Constants.Security.Authorization#ENABLED} to true in
+ *   {@code cdap-site.xml}. When authorization is disabled, an instance of {@link NoOpAuthorizer} is returned.</li>
+ *   <li>The path to the extension jar bundled with all its dependencies is read from the setting
+ *   {@link Constants.Security.Authorization#EXTENSION_JAR_PATH} in cdap-site.xml</li>
+ *   <li>The instantiator reads a fully qualified class name specified as the {@link Attributes.Name#MAIN_CLASS}
+ *   attribute in the extension jar's manifest file. This class must implement {@link Authorizer} and have a default
+ *   constructor.</li>
+ *   <li>During {@link AbstractIdleService#startUp()}, the service creates an instance of of the {@link Authorizer}
+ *   class and also calls its {@link Authorizer#initialize(AuthorizationContext)} method with an
+ *   {@link AuthorizationContext} created using a {@link AuthorizationContextFactory} by providing it a
+ *   {@link Properties} object that is populated with all configuration settings from {@code cdap-site.xml} that have
+ *   keys with the prefix {@link Constants.Security.Authorization#EXTENSION_CONFIG_PREFIX}.</li>
+ *   <li>During {@link AbstractIdleService#shutDown()}, the {@link Authorizer#destroy()} method is invoked, and the
+ *   {@link AuthorizerClassLoader} is closed.</li>
+ * </ul>
+ */
+public class AuthorizationService extends AbstractIdleService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AuthorizationService.class);
+
+  private final CConfiguration cConf;
+  private final boolean authenticationEnabled;
+  private final boolean authorizationEnabled;
+  private final InstantiatorFactory instantiatorFactory;
+  private final AuthorizationContextFactory authorizationContextFactory;
+
+  private File tmpDir;
+  private AuthorizerClassLoader authorizerClassLoader;
+  private Authorizer authorizer;
+
+  public AuthorizationService(CConfiguration cConf, AuthorizationContextFactory authorizationContextFactory) {
```

Comment:
there are no changes in this class other than it not being a ``Supplier`` now, right?